tests/test_config.py

```diff
@@ -135,36 +136,36 @@ def test_config_eol(tmpdir):
         assert cfg.project == u'spam'
 
 
-TYPECHECK_OVERRIDES = [
-    # configuration key, override value, should warn, default type
-    ('master_doc', 123, True, str),
-    ('man_pages', 123, True, list),  # lambda
-    ('man_pages', [], False, list),
-    ('epub_tocdepth', True, True, int),  # child type
-    ('nitpicky', 3, False, bool),  # parent type
-    ('templates_path', (), True, list),  # other sequence, also raises
-]
-if PY2:
-    # Run a check for proper sibling detection in Python 2.  Under py3k, the
-    # default types do not have any siblings.
-    TYPECHECK_OVERRIDES.append(
-            ('html_add_permalinks', 'bar', False, unicode))
-
-def test_gen_check_types():
-    for key, value, should, deftype in TYPECHECK_OVERRIDES:
-        warning = StringIO()
-        app = TestApp(confoverrides={key: value}, warning=warning)
-        app.cleanup()
-
-        real = type(value).__name__
-        msg = ("WARNING: the config value %r has type `%s',"
-               " defaults to `%s.'\n" % (key, real, deftype.__name__))
-        def test():
-            warning_list = warning.getvalue()
-            assert (msg in warning_list) == should, \
-                    "Setting %s to %r should%s raise: %s" % \
-                    (key, value, " not" if should else "", msg)
-        test.description = "test_check_type_%s_on_%s" % \
-                (real, type(Config.config_values[key][0]).__name__)
-
-        yield test
+@with_app(confoverrides={
+        'master_doc': 123,
+        'language': 'foo',
+        'primary_domain': None})
+def test_builtin_conf(app, status, warning):
+    warning = warning.getvalue()
+    assert_in('master_doc', warnings,
```

Comment:
NameError: global name 'warnings' is not defined
https://travis-ci.org/sphinx-doc/sphinx/jobs/79866849